lib/metadata_json_lint.rb

```diff
@@ -98,4 +97,32 @@ def parse(metadata)
     end
   end
   module_function :parse
+
+  def validate_dependencies(deps)
+    dep_names = []
+    deps.each do |dep|
+      if dep_names.include?(dep['name'])
+        puts "Error: duplicate dependencies on #{dep['name']}"
+        @error_state = true
+      end
+      dep_names << dep['name']
+
+      # Open ended dependency
+      # From: https://docs.puppet.com/puppet/latest/reference/modules_metadata.html#best-practice-set-an-upper-bound-for-dependencies
+      next unless dep['version_requirement'].nil? || open_ended?(dep['version_requirement'])
+      puts "Warning: Dependency #{dep['name']} has an open " \
+        "ended dependency version requirement #{dep['version_requirement']}"
+      @error_state = true if options[:strict_dependencies]
```

Comment:
I'd suggest just keeping a local variable with the error state, then return this at the end of the function - the caller can then assign its `error_state` variable to false if this failed. This'll avoid having global state in `@error_state` inside a module.